cmd/soroban-rpc/internal/integrationtest/infrastructure/client.go

```diff
@@ -0,0 +1,196 @@
+package infrastructure
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/creachadair/jrpc2"
+	"github.com/creachadair/jrpc2/jhttp"
+	"github.com/stellar/go/keypair"
+	"github.com/stellar/go/protocols/stellarcore"
+	"github.com/stellar/go/txnbuild"
+	"github.com/stellar/go/xdr"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/methods"
+)
+
+// Client is a jrpc2 client which tolerates errors
+type Client struct {
+	url  string
+	cli  *jrpc2.Client
+	opts *jrpc2.ClientOptions
+}
+
+func NewClient(url string, opts *jrpc2.ClientOptions) *Client {
+	c := &Client{url: url, opts: opts}
+	c.refreshClient()
+	return c
+}
+
+func (c *Client) refreshClient() {
+	if c.cli != nil {
+		c.cli.Close()
+	}
+	ch := jhttp.NewChannel(c.url, nil)
+	c.cli = jrpc2.NewClient(ch, c.opts)
+}
+
+func (c *Client) CallResult(ctx context.Context, method string, params, result any) error {
+	err := c.cli.CallResult(ctx, method, params, result)
+	if err != nil {
+		// This is needed because of https://github.com/creachadair/jrpc2/issues/118
+		c.refreshClient()
+	}
+	return err
+}
+
+func (c *Client) Close() error {
+	return c.cli.Close()
+}
+
+func getTransaction(t *testing.T, client *Client, hash string) methods.GetTransactionResponse {
+	var result methods.GetTransactionResponse
+	for i := 0; i < 60; i++ {
+		request := methods.GetTransactionRequest{Hash: hash}
+		err := client.CallResult(context.Background(), "getTransaction", request, &result)
+		assert.NoError(t, err)
+
+		if result.Status == methods.TransactionStatusNotFound {
+			time.Sleep(time.Second)
+			continue
+		}
+
+		return result
+	}
+	t.Fatal("GetTransaction timed out")
+	return result
+}
+
+func SendSuccessfulTransaction(t *testing.T, client *Client, kp *keypair.Full, transaction *txnbuild.Transaction) methods.GetTransactionResponse {
```

Comment:
can/should these be methods of `Client`?

Reply:
Uhm .. not really since it's pretty opinionated. Sending a transaction (fullstop) probably should.